Failed tests (11210): TestRangeDescriptorSnapshotRace

The following test appears to have failed:

[#11210](https://circleci.com/gh/cockroachdb/cockroach/11210):

```
1      storage/client_test.go:323
I0113 08:53:11.733036 942 stopper.go:264  draining; tasks left:
1      storage/replica.go:1734
I0113 08:53:11.736199 942 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
I0113 08:53:11.736469 942 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestReplicateAfterSplit (0.07s)
	client_raft_test.go:986: Range MaxBytes is not set after snapshot applied
=== RUN   TestRangeDescriptorSnapshotRace
E0113 08:53:11.740753 942 gossip/gossip.go:210  different node IDs were set for the same gossip instance (2147483647, 1)
I0113 08:53:11.741312 942 raft/raft.go:446  [group 1] 1 became follower at term 5
I0113 08:53:11.741571 942 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0113 08:53:11.741705 942 raft/raft.go:526  [group 1] 1 is starting a new election at term 5
I0113 08:53:11.741884 942 raft/raft.go:459  [group 1] 1 became candidate at term 6
I0113 08:53:11.742020 942 raft/raft.go:508  [group 1] 1 received vote from 1 at term 6
I0113 08:53:11.742171 942 raft/raft.go:487  [group 1] 1 became leader at term 6
I0113 08:53:11.743630 942 storage/replica.go:1611  gossiping cluster id  from store 1, range 1
--
I0113 08:53:11.844295 942 raft/raft.go:508  [group 2] 1 received vote from 1 at term 6
I0113 08:53:11.844398 942 raft/raft.go:487  [group 2] 1 became leader at term 6
W0113 08:53:11.848520 942 kv/dist_sender.go:635  failed to invoke BeginTransaction [/Local/Range/RangeDescriptor/"A020",/Min), ConditionalPut [/Local/Range/RangeDescriptor/"A020",/Min), ConditionalPut [/Local/Range/RangeDescriptor/"",/Min), Put [/Meta2/Max,/Min), Put [/Meta2/"A020",/Min), Put [/Meta1/Max,/Min): unexpected value: raw_bytes:"\340u\323\345\003\010\001\022\000\032\001\223\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:0 logical:100 > 
I0113 08:53:11.849291 942 client/db.go:485  failed batch: unexpected value: raw_bytes:"\340u\323\345\003\010\001\022\000\032\001\223\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:0 logical:100 > 
I0113 08:53:11.851575 942 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- FAIL: TestRangeDescriptorSnapshotRace (0.12s)
	client_raft_test.go:1058: storage/replica_command.go:1339: split at key "A020" failed: unexpected value: raw_bytes:"\340u\323\345\003\010\001\022\000\032\001\223\"\006\010\001\020\001\030\001(\002" timestamp:<wall_time:0 logical:100 > 
	<autogenerated>:28: prior leak detected, leaktest disabled
=== RUN   TestRaftAfterRemoveRange
E0113 08:53:11.856138 942 gossip/gossip.go:210  different node IDs were set for the same gossip instance (2147483647, 1)
I0113 08:53:11.856653 942 raft/raft.go:446  [group 1] 1 became follower at term 5
I0113 08:53:11.856908 942 raft/raft.go:234  [group 1] newRaft 1 [peers: [1], term: 5, commit: 10, applied: 10, lastindex: 10, lastterm: 5]
I0113 08:53:11.857044 942 raft/raft.go:526  [group 1] 1 is starting a new election at term 5
I0113 08:53:11.857172 942 raft/raft.go:459  [group 1] 1 became candidate at term 6
I0113 08:53:11.857302 942 raft/raft.go:508  [group 1] 1 received vote from 1 at term 6
I0113 08:53:11.857445 942 raft/raft.go:487  [group 1] 1 became leader at term 6
--
I0113 08:53:20.158226 942 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestLogSplits (1.02s)
	<autogenerated>:28: prior leak detected, leaktest disabled
=== RUN   Example_rebalancing
--- PASS: Example_rebalancing (0.73s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	19.323s
=== RUN   TestBatchBasics
I0113 08:53:22.526519 1006 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I0113 08:53:22.530691 1006 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I0113 08:53:22.537541 1006 storage/engine/rocksdb.go:137  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.01s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/config/config.go
+++ b/config/config.go
@@ -70,6 +70,9 @@ var (
 // that re-enables this splitting.
 func TestingDisableTableSplits() func() {
 	testingLock.Lock()
+	if testingDisableTableSplits {
+		log.Fatal("TestingDisableTableSplits called twice without cleaning up")
+	}
 	testingDisableTableSplits = true
 	testingLock.Unlock()
 	return func() {

--- a/storage/client_raft_test.go
+++ b/storage/client_raft_test.go
@@ -791,9 +791,7 @@ func TestProgressWithDownNode(t *testing.T) {
 func TestReplicateAddAndRemove(t *testing.T) {
 	defer leaktest.AfterTest(t)
 
-	// Run the test twice, once adding the replacement before removing
-	// the downed node, and once removing the downed node first.
-	for _, addFirst := range []bool{true, false} {
+	testFunc := func(addFirst bool) {
 		mtc := startMultiTestContext(t, 4)
 		defer mtc.Stop()
 
@@ -872,6 +870,10 @@ func TestReplicateAddAndRemove(t *testing.T) {
 			t.Fatalf("expected replica IDs to be %v but got %v", expected, replicaIDsByStore)
 		}
 	}
+	// Run the test twice, once adding the replacement before removing
+	// the downed node, and once removing the downed node first.
+	testFunc(true)
+	testFunc(false)
 }
 
 // TestRaftHeartbeats verifies that coalesced heartbeats are correctly

--- a/storage/client_test.go
+++ b/storage/client_test.go
@@ -276,8 +276,8 @@ func (m *multiTestContext) Stop() {
 		for _, s := range m.engineStoppers {
 			s.Stop()
 		}
-		close(done)
 		m.reenableTableSplits()
+		close(done)
 	}()
 
 	select {

